docs/tooling/unity-sdk/guides/convert.md

```diff
@@ -0,0 +1,231 @@
+annotated, it will be skipped when converting from C# to Cadence.
+
+You can also use the Name parameter to account for differences in field naming.  In the above example we mapped the Cadence "i32" field to the C# "i" field.  The Name property is optional and
+it will use the field name if no Name property is given.
+
+***Note:  The CadenceType annotation on C# classes/structs ([Cadence(CadenceType="A.XXX.CCC.TestStruct")] in the above example) is ignored when converting from Cadence to C#.
+Convert.FromCadence will populate all fields that have matching names/types regardless of the type of Cadence struct that is being converted.  This allows you to convert a Cadence struct defined
+int a Cadence script into C# even if you do not know what the Cadence type is.***
+
+The class annotation (A.XXX.CCC.TestStruct) is required when converting from C# to Cadence. XXX should be the address of the account that contains the contract where the struct is 
+defined, without a leading "0x".  The CCC is the name of the contract.
+
+Here's an example using the NBA TopShot contract (https://github.com/dapperlabs/nba-smart-contracts/blob/master/contracts/TopShot.cdc) on TestNet, defining a class to hold the Play struct:
+
+```csharp
+[Cadence(CadenceType="A.877931736ee77cff.TopShot.Play")]
+public class Play
+{
+    [Cadence(CadenceType="UInt32")]
+    public UInt32 playID; 
+    [Cadence(CadenceType="{String:String}")]
+    public Dictionary<String, String> metadata;
+}
+```
+
+## Structs inside Structs
+
+If a Cadence struct contains another struct, the field should be annotated as a "Struct".  Given the following Cadence:
+
+```cadence
+pub struct Other {
+    pub let i: Int16
+
+    pub init(i:Int16) {
+        self.i=i
+    }
+}
+
+pub struct Test {
+    pub let o : Other
+    
+    pub init(i: Int16) {
+        self.o = Other(i:i)
+    }
+}
+```
+
+you could use this C#:
+
+```csharp
+[Cadence(CadenceType="A.xxx.ccc.Nested")]
+public class Other
+{
+    [Cadence(CadenceType="Int16")]
+    public Int16 i;
+}
+
+[Cadence(CadenceType="A.xxx.ccc.Test")]
+public class Test
+{
+    [Cadence(CadenceType="Struct")]
+    public Other o;
+} 
+```
+
+If you have a Cadence Test struct you can convert into C# using:
+```csharp
+TestStruct ts = Convert.FromCadence<TestStruct>(cadence);
+```
+
+If you have a C# Test object, you can convert to Cadence using:
+```csharp
+CadenceBase cb = Convert.ToCadence(ts, "Struct");
+```
+
+## Optionals
+
+Cadence optionals are indicated by appending a ?.  For instance the Cadence type Int16? can contain either an Int16 value or nil.  If the C# type is a reference type, no additional work is required.
+For instance a Cadence String? will have the C# equivalent type of String.  This is because the C# String is a reference type, which can natively be set to null.  On the other hand, the Cadence Int16? requires the C# type Int16? which wraps type the value type Int16 in a Nullable<>.
```

Comment:
```suggestion
For instance a Cadence String? will have the C# equivalent type of String.  This is because the C# String is a reference type, which can natively be set to null.  On the other hand, the Cadence Int16? requires the C# type Int16? which wraps type the value type Int16 in a `Nullable<>`.
```